api/src/main/java/org/apache/iceberg/StructTransform.java

```diff
@@ -125,6 +151,11 @@ public int hashCode() {
     return Arrays.hashCode(transformedTuple);
   }
 
+  /** Return the schema of transformed result (a flattened structure) */
+  public Schema resultSchema() {
```

Comment:
My only question here is why we want to produce this as a Schema? Wouldn't matching the List<Transforms> fit better here? Then we wouldn't have to create artificial fieldId's. If we mostly need Schema out of this api though that would make sense though.

Reply:
Do you mean `List<NestedField>` or `List<Type>`? the former still requires field ids. 

For my use case (Flink sink shuffling), we want to use `StructLikeComparator` to compare the `SortKey` values. Hence need the result schema of the `SortKey/StructTransform`.

Admittedly, the field ids and names here are very meaningful. With `List<Type>`, I can construct a schema with artificial field ids and names at the caller side. so it is also workable.

Reply:
following on that logic, let me think more if I really need this change. Maybe I can just extract `List<Type> resultTypes` from the `List<FieldTransform> fieldTransforms`

Reply:
> Maybe I can just extract List<Type> resultTypes from the List<FieldTransform> fieldTransforms

Yeah, this should be doable. I can construct the schema with artificial field ids and names at the caller side